src/views/Application.vue

```diff
@@ -3,23 +3,35 @@
     <RefuseUI />
   </div>
   <div v-else>
-    <ApplicationUI />
+    <ApplicationUI v-if="this.user.role == 'hacker'" />
+    <MentorApp v-else-if="this.user.role == 'mentor'" />
+    <VolunteerApp v-else-if="this.user.role == 'volunteer'" />
   </div>
 </template>
 
 <script>
 import ApplicationUI from "@/components/hacker/ApplicationUI";
+import MentorApp from "@/components/mentor/MentorApp";
+import VolunteerApp from "@/components/volunteer/VolunteerApp";
 import RefuseUI from "@/components/reject/RefuseUI";
 import { db } from "@/firebase/init";
 
 export default {
+  computed: {
+    user() {
+      return this.$store.state.user;
+    }
+  },
   data() {
     return {
       date: new Date().toISOString().substr(0, 10),
       deadline: ""
     };
   },
   async mounted() {
+    if (this.user.role == "sponsor" || this.user.role == "admin") {
+      this.$router.push({ name: "home" });
+    }
```

Comment:
this works, but reimplement this within router so that it is cleaner!

Reply:
@jfeng5529

Reply:
Just fixed it!